src/auto_note.cpp

```diff
@@ -0,0 +1,454 @@
+#include "auto_note.h"
+
+#include <iostream>
+
+
+#include "game.h"
+#include "json.h"
+#include "string_formatter.h"
+#include "color.h"
+#include "cursesdef.h"
+#include "input.h"
+#include "options.h"
+#include "cata_utility.h"
+#include "output.h"
+#include "debug.h"
+#include "filesystem.h"
+#include "translations.h"
+#include "avatar.h"
+#include "map_extras.h"
+
+namespace auto_notes
+{
+std::string auto_note_settings::build_save_path() const
+{
+    return g->get_player_base_save_path() + ".ano.json";
+}
+
+void auto_note_settings::clear()
+{
+    this->autoNoteEnabled.clear();
```

Comment:
```suggestion
    autoNoteEnabled.clear();
```

There are many more unnecessary `this->` in the code. `this->` is almost never needed and purely redundant.